### ISSUE TITLE
[Issue]: Generating incorrect album art for albums with the same name

### ISSUE THREAD

**Issue by Coloradohusky** [jellyfin/jellyfin#8194]
### Please describe your bug

Running the latest stable version of Jellyfin on Windows 10, EXE not Docker. I currently use MusicBrainz to name my albums, with the rough schema of [album artist]/[album](first 8 digits of MusicBrainz Album ID)/[song], which looks like this:
![Screenshot 2022-07-31 215546](https://user-images.githubusercontent.com/45430066/182075177-3b5aae0c-f343-4fe5-98df-130098608b60.png)
When I scan these albums in with a fresh install of Jellyfin, I get this:
![Screenshot 2022-07-31 215714](https://user-images.githubusercontent.com/45430066/182075298-8952ad5d-ce31-476b-a5c6-7e1186feb9c2.png)
All self-titled albums (except for Teal) have the Red Album art. If we look in Foobar2000, all of them have the correct album art embedded in the files.
![Screenshot 2022-07-31 215822](https://user-images.githubusercontent.com/45430066/182075405-14c30f3c-e278-49cb-adc5-8c875b2738ea.png)
Looking at the debug logs it first:
1. Correctly generates the Teal Album art.
2. Correctly generates the Red Album art.
3. Somehow generates the Red Album art... for the Green Album, and for all subsequent albums named 'Weezer'.

Teal:
```
[2022-07-31 19:36:28.847 -07:00] [DBG] [45] MediaBrowser.Providers.Music.AlbumMetadataService: Running "MusicAlbumImageProvider" for "E:\WEEZER TEST\Weezer (30b7c1cf)"
[2022-07-31 19:36:29.036 -07:00] [DBG] [45] Emby.Server.Implementations.Data.SqliteItemRepository: "GetItemList" query time (slow): 186.526ms. Query: "select type,data,EndDate,ChannelId,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,TrailerTypes,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,SeriesPresentationUniqueKey,OwnerId from TypedBaseItems A where ParentId=@ParentId AND Images like '%Primary%' ORDER BY IsFolder ASC,SortName ASC LIMIT 1"
[2022-07-31 19:36:29.046 -07:00] [DBG] [45] MediaBrowser.Providers.Manager.ProviderManager: Saving image to "C:\ProgramData\Jellyfin\Server\metadata\library\19\192891c26b77b3cbb80f7f8565985e95\folder.jpg"
```
Red:
```
[2022-07-31 19:36:32.229 -07:00] [DBG] [42] MediaBrowser.Providers.Music.AlbumMetadataService: Running "MusicAlbumImageProvider" for "E:\WEEZER TEST\Weezer (3f9068bb)"
[2022-07-31 19:36:32.269 -07:00] [DBG] [42] Emby.Server.Implementations.Data.SqliteItemRepository: "GetItemList" query time (slow): 36.2726ms. Query: "select type,data,EndDate,ChannelId,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,TrailerTypes,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,SeriesPresentationUniqueKey,OwnerId from TypedBaseItems A where ParentId=@ParentId AND Images like '%Primary%' ORDER BY IsFolder ASC,SortName ASC LIMIT 1"
[2022-07-31 19:36:32.283 -07:00] [DBG] [42] MediaBrowser.Providers.Manager.ProviderManager: Saving image to "C:\ProgramData\Jellyfin\Server\metadata\library\92\92b84af1c6c47fdf9c4651e5b80e3ace\folder.jpg"
```
Green:
```
[2022-07-31 19:36:32.939 -07:00] [DBG] [42] MediaBrowser.Providers.Music.AlbumMetadataService: Running "MusicAlbumImageProvider" for "E:\WEEZER TEST\Weezer (44447531)"
[2022-07-31 19:36:33.086 -07:00] [DBG] [42] Emby.Server.Implementations.Data.SqliteItemRepository: "GetItemList" query time (slow): 145.0798ms. Query: "select type,data,EndDate,ChannelId,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,TrailerTypes,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,SeriesPresentationUniqueKey,OwnerId from TypedBaseItems A where ParentId=@ParentId AND Images like '%Primary%' ORDER BY IsFolder ASC,SortName ASC LIMIT 1"
[2022-07-31 19:36:33.098 -07:00] [DBG] [42] MediaBrowser.Providers.Manager.ProviderManager: Saving image to "C:\ProgramData\Jellyfin\Server\metadata\library\04\0480b599c1ed498933ce64105411cbda\folder.jpg"
```
The SQL data for all three:
```
MediaBrowser.Controller.Entities.Audio.MusicAlbum			E:\WEEZER TEST\Weezer (30b7c1cf)								0	Weezer					2019-01-24 00:00:00Z	2019	Alternative Rock	weezer		21843068928	2022-08-01 01:22:54.1621359Z	0001-01-01 00:00:00Z						2022-08-01 02:36:29.1025907Z	2022-08-01 02:36:29.1026261Z	0		Atlantic|Crush Music				1	0	Music	26299946dd4299ccf606e1c9fe660b91			weezer	192891c26b77b3cbb80f7f8565985e95					0		Weezer-Weezer							%MetadataPath%\library\19\192891c26b77b3cbb80f7f8565985e95\folder.jpg*637949181890495046*Primary*600*600*eRCJFDxuO?NxAB3USfrrt7-Ctla~jFjsV@#;s:S~NGX4NZRjnixaoM					Weezer	Weezer							
MediaBrowser.Controller.Entities.Audio.MusicAlbum			E:\WEEZER TEST\Weezer (3f9068bb)								0	Weezer					2008-06-03 00:00:00Z	2008	Alternative Rock|Rock	weezer		40123297792	2022-08-01 01:22:48.9185011Z	0001-01-01 00:00:00Z						2022-08-01 02:36:32.2886211Z	2022-08-01 02:36:32.2886578Z	0						1	0	Music	26299946dd4299ccf606e1c9fe660b91			weezer	92b84af1c6c47fdf9c4651e5b80e3ace					0		Weezer-Weezer							%MetadataPath%\library\92\92b84af1c6c47fdf9c4651e5b80e3ace\folder.jpg*637949181922869609*Primary*1200*1200*e=J;Y~s:n%xGof{~niWBjFWBw{o1ayfQj[X8j[jsoLj[jZWVayayfP					Weezer	Weezer							
MediaBrowser.Controller.Entities.Audio.MusicAlbum			E:\WEEZER TEST\Weezer (44447531)								0	Weezer					2001-01-01 00:00:00Z	2001	Alternative Rock|Rock	weezer		17131200512	2022-08-01 01:22:59.4542423Z	0001-01-01 00:00:00Z						2022-08-01 02:36:33.1258378Z	2022-08-01 02:36:33.1258725Z	0		Geffen Records				1	0	Music	26299946dd4299ccf606e1c9fe660b91			weezer	0480b599c1ed498933ce64105411cbda					0		Weezer-Weezer							%MetadataPath%\library\04\0480b599c1ed498933ce64105411cbda\folder.jpg*637949181931039555*Primary*1200*1200*e=J;Y~s:n%xGof{~niWBjFWBw{o1ayfQj[X8j[jsoLj[jZWVayayfP					Weezer	Weezer							
```

Personally, I'm highly suspicious that the SQL query of "GetItemList" is pulling the wrong album. Tried running the query in 'DB Browser for SQLite', but I don't know how to use BLOBs in queries.

[MusicAlbumImageProvider.cs](https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Images/MusicAlbumImageProvider.cs)
[AlbumMetadataService.cs](https://github.com/jellyfin/jellyfin/blob/ef0708d876434a99ec647473c37295fab45a35fb/MediaBrowser.Providers/Music/AlbumMetadataService.cs)
[BaseFolderImageProvider.cs](https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Images/BaseFolderImageProvider.cs) (lines 33-44 seem to be the SQL query)

**TLDR:** I expect Jellyfin to pull album art from the embedded images. It works for some albums, but doesn't work if the albums have the same name, instead giving them all the same album art.

### Jellyfin Version

10.8.1

### Environment

```markdown
- OS: Windows
- Virtualization: None
- Clients: N/A
- Browser: N/A
- FFmpeg Version: N/A
- Playback Method: N/A
- Hardware Acceleration: N/A
- Plugins: Standard plugins
- Reverse Proxy: None
- Base URL: None
- Networking: Host
- Storage: Local
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Coloradohusky** [comment]
Still appears to be happening on 10.8.7

**Comment by Coloradohusky** [comment]
Still an issue on 10.8.9

**Comment by Coloradohusky** [comment]
Still an issue as of 10.8.10.
